### PR TITLE
NAS-101507 / 11.3 / Improve jail updates

### PR DESCRIPTION
**Pull request by sonicaj**
This commit introduces following changes:
1) A flag to update pkg repositories in the jail and then upgrade all existing pkgs in the jail.
2) Ability to update jail itself instead of only the software inside them for plugins.
3) Allow running pkg/exec for ALL jails.

Ticket: #NAS-101507

### CODE REVIEW

File: iocage_cli/update.py [hunk 1]
```diff
@@ -29,11 +29,27 @@
...
+    help='Run freebsd-update to update a specified '
+    'jail to the latest patch level.'
+)
+@click.option(
+    '--pkgs/--nopkgs', default=False,
+    help='Decide whether or not to update the pkg repositories and '
+         'all installed packages in jail.'
+)
+@click.option(
+    '--plugins/--noplugins', '-P/-NP', default=True,
```
Comment (skarekrow) [inline iocage_cli/update.py]: This doesn't make sense, they're calling update, whatever the type is should always be updated.

File: iocage_cli/update.py [hunk 2]
```diff
@@ -29,11 +29,27 @@
...
+    help='Decide whether or not to update the plugin.'
+)
+@click.option(
+    '--jailversion/--nojailversion', '-J/-NJ', default=True,
+    help='Decide whether or not to update the plugin.'
```
Comment (skarekrow) [inline iocage_cli/update.py]: Same here

File: iocage_cli/update.py [hunk 3]
```diff
@@ -29,11 +29,27 @@
...
-@click.argument("jail", required=True)
-def cli(jail):
+    name='update',
+    help='Run freebsd-update to update a specified '
+    'jail to the latest patch level.'
+)
+@click.option(
+    '--pkgs/--nopkgs', default=False,
+    help='Decide whether or not to update the pkg repositories and '
+         'all installed packages in jail.'
```
Comment (skarekrow) [inline iocage_cli/update.py]: We should make this true if the jail in question is a plugin jail, this flag should only be false for plugins if they specifically pass --nopkgs

**Review by skarekrow — CHANGES_REQUESTED** [review]

File: iocage_cli/update.py (hunk 3 above)
Comment (sonicaj) [reply iocage_cli/update.py]: I am not sure but to be sure, is there potential for conflict between plugin updates and pkg updates in the same jail ?

Comment (skarekrow) [reply iocage_cli/update.py]: They're the same thing. Plugin updates currently are only pkg's + snapshot. So if they specifically disable that, all we should do in the plugin update process is snapshot and then update the world.

File: iocage_cli/update.py (hunk 1 above)
Comment (sonicaj) [reply iocage_cli/update.py]: There's a usecase, say latest plugin update version has bugs. So they would only like to update freebsd jail. In that case they are able to do so. We default to `True` so this is executed in most cases.

Comment (skarekrow) [reply iocage_cli/update.py]: That isn't necessary, the pkg flag would control that. The jail should always be updated. It's why they're calling it :P

**Review by skarekrow — APPROVED** [review]

**Pull request merged**